src/Controller/Component/SearchComponent.php

```diff
@@ -99,8 +99,13 @@ public function beforeRender()
 
         $controller = $this->getController();
         try {
+            // check to ensure modelclass is set before loading it
+            $model_class = $this->getConfig('modelClass');
+            if (!$model_class) {
```

Comment:
Doesnt this mean it skips for the default model, that loadModel() would actually load with empty param?
Wouldnt it be better to check for strict empty string and skip in that case?

```suggestion
            if ($modelClass === '') {
```

Reply:
Also, why would the component be added to controllers that do not need them?

Reply:
Loading this component and others in a Parent Controller class that is extended by all other controllers.